+ source /root/mugen/libs/locallibs/common_lib.sh
++ python3 --version
++ '[' 0 -eq 0 ']'
++ source /root/mugen/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start to prepare the test environment.'
+ message='Start to prepare the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to prepare the test environment.'
Thu Jun 22 09:29:42 2023 - INFO  - Start to prepare the test environment.
+ OLD_LANG=en_US.UTF-8
+ export LANG=en_US.UTF-8
+ LANG=en_US.UTF-8
+ DNF_INSTALL 'bacula-client mysql5-server bacula-common tar'
+ pkgs='bacula-client mysql5-server bacula-common tar'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen/libs/locallibs/rpm_manage.py install --pkgs 'bacula-client mysql5-server bacula-common tar' --node 1 --tempfile ''
+ tmpfile2='Last metadata expiration check: 0:20:26 ago on Thu 22 Jun 2023 09:09:38 AM CST.
No match for argument: mysql5-server
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: Unable to find a match: mysql5-server'
+ '[' -z '' ']'
+ tmpfile='Last metadata expiration check: 0:20:26 ago on Thu 22 Jun 2023 09:09:38 AM CST.
No match for argument: mysql5-server
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: Unable to find a match: mysql5-server'
+ tar -zxvf common/test.tar.gz
config/
config/a.config
config/4.conf
config/3.conf
config/2.conf
config/6.conf
config/5.conf
config/1.conf
config/e.conf
+ systemctl restart mysqld
Failed to restart mysqld.service: Unit mysqld.service not found.
+ mkdir tmp
+ /usr/libexec/bacula/create_mysql_database
oe_test_bacula_dbcheck_01.sh: line 30: /usr/libexec/bacula/create_mysql_database: No such file or directory
+ /usr/libexec/bacula/make_mysql_tables
oe_test_bacula_dbcheck_01.sh: line 31: /usr/libexec/bacula/make_mysql_tables: No such file or directory
+ alternatives --set libbaccats.so /usr/lib64/libbaccats-mysql.so
cannot access /var/lib/alternatives/libbaccats.so: No such file or directory
+ sed -i 's\dbuser = "bacula"\dbuser = "root"\g' /etc/bacula/bacula-dir.conf
sed: can't read /etc/bacula/bacula-dir.conf: No such file or directory
+ systemctl start bacula-dir.service bacula-sd.service bacula-fd.service
Failed to start bacula-dir.service: Unit bacula-dir.service not found.
Failed to start bacula-sd.service: Unit bacula-sd.service not found.
Failed to start bacula-fd.service: Unit bacula-fd.service not found.
+ LOG_INFO 'End to prepare the test environment.'
+ message='End to prepare the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to prepare the test environment.'
Thu Jun 22 09:30:06 2023 - INFO  - End to prepare the test environment.
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start to run test.'
+ message='Start to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to run test.'
Thu Jun 22 09:30:07 2023 - INFO  - Start to run test.
+ expect
couldn't execute "dbcheck": no such file or directory
    while executing
"spawn dbcheck -v -B -C MyCatalog -d 11 tmp/back_1 bacula root ""  127.0.0.1  3306 ssh"
+ grep 'Verbose is ' tmp/4.txt
+ CHECK_RESULT 1 0 0 'test dbcheck -v -B -C -d failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='test dbcheck -v -B -C -d failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'test dbcheck -v -B -C -d failed'
+ LOG_ERROR 'test dbcheck -v -B -C -d failed'
+ message='test dbcheck -v -B -C -d failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'test dbcheck -v -B -C -d failed'
Thu Jun 22 09:30:08 2023 - ERROR - test dbcheck -v -B -C -d failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_bacula_dbcheck_01.sh line 46'
+ message='oe_test_bacula_dbcheck_01.sh line 46'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_bacula_dbcheck_01.sh line 46'
Thu Jun 22 09:30:09 2023 - ERROR - oe_test_bacula_dbcheck_01.sh line 46
+ '[' 0 -eq 1 ']'
+ return 0
+ expect
couldn't execute "dbcheck": no such file or directory
    while executing
"spawn dbcheck -v -B -C MyCatalog -dt -d 11 tmp/back_1 bacula root ""  127.0.0.1  3306 ssh"
+ grep 'Verbose is ' tmp/5.txt
+ CHECK_RESULT 1 0 0 'test dbcheck -dt failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='test dbcheck -dt failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'test dbcheck -dt failed'
+ LOG_ERROR 'test dbcheck -dt failed'
+ message='test dbcheck -dt failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'test dbcheck -dt failed'
Thu Jun 22 09:30:10 2023 - ERROR - test dbcheck -dt failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_bacula_dbcheck_01.sh line 53'
+ message='oe_test_bacula_dbcheck_01.sh line 53'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_bacula_dbcheck_01.sh line 53'
Thu Jun 22 09:30:10 2023 - ERROR - oe_test_bacula_dbcheck_01.sh line 53
+ '[' 0 -eq 1 ']'
+ return 0
+ expect
couldn't execute "dbcheck": no such file or directory
    while executing
"spawn dbcheck -f -v -B -C MyCatalog -dt -d 11 tmp/back_1 bacula root ""  127.0.0.1  3306 ssh"
+ grep 'Verbose is ' tmp/6.txt
+ CHECK_RESULT 1 0 0 'test dbcheck -f failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='test dbcheck -f failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'test dbcheck -f failed'
+ LOG_ERROR 'test dbcheck -f failed'
+ message='test dbcheck -f failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'test dbcheck -f failed'
Thu Jun 22 09:30:11 2023 - ERROR - test dbcheck -f failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_bacula_dbcheck_01.sh line 60'
+ message='oe_test_bacula_dbcheck_01.sh line 60'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_bacula_dbcheck_01.sh line 60'
Thu Jun 22 09:30:12 2023 - ERROR - oe_test_bacula_dbcheck_01.sh line 60
+ '[' 0 -eq 1 ']'
+ return 0
+ expect
couldn't execute "dbcheck": no such file or directory
    while executing
"spawn dbcheck -b -v -d 21 tmp/back_1 bacula root ""  127.0.0.1  3306 ssh"
+ grep 'Found 0 bad Path records.' tmp/7.txt
+ CHECK_RESULT 1 0 0 'test dbcheck -b failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='test dbcheck -b failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'test dbcheck -b failed'
+ LOG_ERROR 'test dbcheck -b failed'
+ message='test dbcheck -b failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'test dbcheck -b failed'
Thu Jun 22 09:30:13 2023 - ERROR - test dbcheck -b failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_bacula_dbcheck_01.sh line 67'
+ message='oe_test_bacula_dbcheck_01.sh line 67'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_bacula_dbcheck_01.sh line 67'
Thu Jun 22 09:30:14 2023 - ERROR - oe_test_bacula_dbcheck_01.sh line 67
+ '[' 0 -eq 1 ']'
+ return 0
+ dbcheck -c /etc/bacula/bacula-dir.conf -f -v -B -C MyCatalog -dt -d 11 tmp/back_1 bacula root '' 127.0.0.1 3306 ssh
+ grep db_name=bacula
+ CHECK_RESULT 1 0 0 'test dbcheck -c failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='test dbcheck -c failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'test dbcheck -c failed'
+ LOG_ERROR 'test dbcheck -c failed'
+ message='test dbcheck -c failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'test dbcheck -c failed'
Thu Jun 22 09:30:14 2023 - ERROR - test dbcheck -c failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_bacula_dbcheck_01.sh line 69'
+ message='oe_test_bacula_dbcheck_01.sh line 69'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_bacula_dbcheck_01.sh line 69'
Thu Jun 22 09:30:15 2023 - ERROR - oe_test_bacula_dbcheck_01.sh line 69
+ '[' 0 -eq 1 ']'
+ return 0
+ expect
couldn't execute "dbcheck": no such file or directory
    while executing
"spawn dbcheck -t tmp/back_1 bacula root ""  127.0.0.1  3306 ssh"
+ grep Hello tmp/8.txt
+ CHECK_RESULT 1 0 0 'test dbcheck -t failed'
+ actual_result=1
+ expect_result=0
+ mode=0
+ error_log='test dbcheck -t failed'
+ exit_mode=0
+ '[' -z 1 ']'
+ '[' 0 -eq 0 ']'
+ test 1x '!=' 0x
+ test -n 'test dbcheck -t failed'
+ LOG_ERROR 'test dbcheck -t failed'
+ message='test dbcheck -t failed'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'test dbcheck -t failed'
Thu Jun 22 09:30:16 2023 - ERROR - test dbcheck -t failed
+ (( exec_result++ ))
+ LOG_ERROR 'oe_test_bacula_dbcheck_01.sh line 75'
+ message='oe_test_bacula_dbcheck_01.sh line 75'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'oe_test_bacula_dbcheck_01.sh line 75'
Thu Jun 22 09:30:17 2023 - ERROR - oe_test_bacula_dbcheck_01.sh line 75
+ '[' 0 -eq 1 ']'
+ return 0
+ LOG_INFO 'End to run test.'
+ message='End to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to run test.'
Thu Jun 22 09:30:17 2023 - INFO  - End to run test.
+ CASE_RESULT 0
+ case_re=0
+ test -z 6
+ test 6 -gt 0
+ LOG_ERROR 'failed to execute the case.'
+ message='failed to execute the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level error --message 'failed to execute the case.'
Thu Jun 22 09:30:18 2023 - ERROR - failed to execute the case.
+ exit 6
+ post_test
+ LOG_INFO 'Start to restore the test environment.'
+ message='Start to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to restore the test environment.'
Thu Jun 22 09:30:18 2023 - INFO  - Start to restore the test environment.
+ export LANG=en_US.UTF-8
+ LANG=en_US.UTF-8
+ systemctl stop bacula-dir.service bacula-sd.service bacula-fd.service
Failed to stop bacula-dir.service: Unit bacula-dir.service not loaded.
Failed to stop bacula-sd.service: Unit bacula-sd.service not loaded.
Failed to stop bacula-fd.service: Unit bacula-fd.service not loaded.
+ sed -i 's\dbuser = "root"\dbuser = "bacula"\g' /etc/bacula/bacula-dir.conf
sed: can't read /etc/bacula/bacula-dir.conf: No such file or directory
+ alternatives --set libbaccats.so /usr/lib64/libbaccats-postgresql.so
cannot access /var/lib/alternatives/libbaccats.so: No such file or directory
+ /usr/libexec/bacula/drop_mysql_tables
oe_test_bacula_dbcheck_01.sh: line 85: /usr/libexec/bacula/drop_mysql_tables: No such file or directory
+ /usr/libexec/bacula/drop_mysql_database
oe_test_bacula_dbcheck_01.sh: line 86: /usr/libexec/bacula/drop_mysql_database: No such file or directory
+ systemctl stop mysqld
Failed to stop mysqld.service: Unit mysqld.service not loaded.
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z Last metadata expiration check: 0:20:26 ago on Thu 22 Jun 2023 09:09:38 AM CST.
No match for argument: mysql5-server
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: Unable to find a match: mysql5-server ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile 'Last metadata expiration check: 0:20:26 ago on Thu 22 Jun 2023 09:09:38 AM CST.
No match for argument: mysql5-server
Package tar-2:1.34-4.oe2303.riscv64 is already installed.
Error: Unable to find a match: mysql5-server'
Traceback (most recent call last):
  File "/root/mugen/libs/locallibs/rpm_manage.py", line 173, in <module>
    exitcode = rpm_remove(args.node, args.pkgs, args.tempfile)
  File "/root/mugen/libs/locallibs/rpm_manage.py", line 144, in rpm_remove
    with open(tmpfile, "r") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'Last metadata expiration check: 0:20:26 ago on Thu 22 Jun 2023 09:09:38 AM CST.\nNo match for argument: mysql5-server\nPackage tar-2:1.34-4.oe2303.riscv64 is already installed.\nError: Unable to find a match: mysql5-server'
+ '[' 0 -ne 0 ']'
+ rm -rf config/ tmp/ '/var/lib/mysql/*' '/var/spool/bacula/*'
+ LOG_INFO 'End to restore the test environment.'
+ message='End to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to restore the test environment.'
Thu Jun 22 09:30:22 2023 - INFO  - End to restore the test environment.
